stacks_queues/queue.py

```diff
@@ -23,39 +27,68 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException('Queue is full!')
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+        self.store[self.rear] = element #insert where rear and front were pointing at first. But rear is the one moving to indicate the position of insert
+        self.rear = (self.rear + 1) % self.buffer_size #the next insert position
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException
+        front_element = self.store[self.front]
+        self.store[self.front] = None
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return front_element
+        # check if empty, if so raise an exception
+        # find and store the front element
+        # move front to the next index
+        # return the old front elements
+        # front and rear are indexes, like 0, 1
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
```

Comment:
👀 This does indicate the queue is empty, but be careful because it can actually also indicate that the queue might be full. Consider using `self.size` or checking what the value of `front` and `rear` are